test/core/libp2p/transport/transport_parser_test.cpp

```diff
@@ -0,0 +1,63 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "libp2p/transport/impl/transport_parser.hpp"
+#include <gtest/gtest.h>
+#include <outcome/outcome.hpp>
+#include <testutil/outcome.hpp>
+
+using libp2p::multi::Multiaddress;
+using libp2p::transport::SupportedProtocol;
+using libp2p::transport::TransportParser;
+
+using ::testing::Test;
+
+struct TransportParserTest : public Test {
+  TransportParserTest()
+      : addr_{Multiaddress::create("/ip4/127.0.0.1/tcp/5050").value()},
+        unsupported_addr_{
+            Multiaddress::create("/ipfs/QmcgpsyWgH8Y8ajJz1Cu72KnS5uo2Aa2LpzU7kinSupNKC/tcp/1234").value()} {}
```

Comment:
please create Parametrized test, and supply some set of valid and invalid multiaddresses.

include ip4/tcp and ip6/tcp